lib/block/blockupload.js

```diff
@@ -0,0 +1,203 @@
+            uploadFile.fileSize = size;
+        }
+
+        const contentType = options.headers && options.headers['content-type'];
+        const transferAsset = new TransferAsset(source, target, {
+            acceptRanges: true,
+            metadata: new AssetMetadata(uploadFile.filePath, contentType, uploadFile.fileSize)
+        });
+
+        const uploadURIs = uploadFile.fileUrl;
+        const minPartSize = uploadFile.minPartSize || 10;
+        const maxPartSize = uploadFile.maxPartSize;
+        
+        if(typeof uploadURIs === 'object' && Array.isArray(uploadURIs)) {
+            transferAsset.multipartTarget = new AssetMultipart(
+                uploadURIs,
+                minPartSize,
+                maxPartSize,
+                transferAsset.target.multipartHeaders
+            );
+        }
+
+        yield transferAsset;
+    }
+}
+
+class BlockUpload extends EventEmitter {
+
+    /**
+     * @typedef {Object} UploadFile
+     * @property {String} fileUrl AEM url where to upload the file
+     * @property {Number} fileSize Size of the file to upload
+     * @property {String} filePath Path on the local disk to upload
+     * @property {Blob} [blob] Browser blob to upload (instead of fileUrl)
+     * @property {Boolean} [createVersion=false] Create version on duplicates
+     * @property {String} [versionLabel] Version label to apply to the created/updated file
+     * @property {String} [versionComment] Version comment to apply to the created/updated file
+     * @property {Boolean} [replace=false] True if the existing file should be replaced
+     */
+    /**
+     * @typedef {Object} BlockUploadOptions
+     * @property {UploadFile[]} uploadFiles List of files that will be uploaded to the target URL. 
+     * @property {*} headers HTTP headers that will be included in each request sent to AEM
+     * @property {Boolean} concurrent If true, multiple files in the supplied list of upload files will transfer simultaneously. If false, only one file will transfer at a time, and the next file will not begin transferring until the current file finishes.
+     * @property {Number} maxConcurrent Maximum number of concurrent HTTP requests that are allowed
+     * @property {Number} [preferredPartSize] Preferred part size
+     */
+    /**
+     * Upload files to AEM
+     * 
+     * @param {BlockUploadOptions} options AEM upload options
+     */
+    async uploadFiles(options = {}) {
+        const preferredPartSize = options.preferredPartSize || DEFAULT_PART_SIZE;
+        const maxConcurrent = (options.concurrent && options.maxConcurrent) || DEFAULT_MAX_CONCURRENCY;
+        const retryOptions = {
+            retryMaxCount: options.retryCount || DEFAULT_RETRY_COUNT,
+            timeout: options.timeout,
+            retryAllErrors: options.retryAllErrors,
+            retryMaxDuration: options.retryMaxDuration,
+            retryInitialDelay: options.retryInitialDelay,
+            retryBackoff: options.retryBackoff,
```

Comment:
@pheenomenon we need to pass all the retry options through since any of them could be used. for example. the sdk uses maxRetryDuration: https://github.com/adobe/asset-compute-pipeline/blob/master/lib/storage/http.js#L66

Should we just pass all options through to  `Transfer` at this point?

Reply:
yep, It makes sense to pass through all the retry options, given [transfer.js is using the same retry wrapper](https://github.com/adobe/node-httptransfer/blob/master/lib/functions/transfer.js#L18) which would then adapt accordingly.